test/e2e/pkg/grammar/abci_grammar.md

```diff
@@ -18,8 +18,11 @@ ConsensusHeight : ConsensusRounds FinalizeBlock Commit | FinalizeBlock Commit ;
 ConsensusRounds : ConsensusRound | ConsensusRound ConsensusRounds ;
 ConsensusRound : Proposer | NonProposer ; 
 
-Proposer : PrepareProposal | PrepareProposal ProcessProposal ; 
-NonProposer: ProcessProposal ;
+Proposer : GotVotes | ProposerSimple | Extend | GotVotes ProposerSimple | GotVotes Extend | ProposerSimple Extend | GotVotes ProposerSimple Extend ; 
```

Comment:
Is this line considering the case where:
* no votes are received
* no prepare proposal is received (as a proposer)
* no extend vote is received

Reply:
Do you mean all three together?

Reply:
Yes, the original grammar is:
```abnf
proposer = *got-vote [prepare-proposal [process-proposal]] [extend]
```
so all three parts are optional.

Also, I just realized the same applies to `NonProposer`

Reply:
This is actually covered with this rule: 

`ConsensusHeight : ConsensusRounds FinalizeBlock Commit | FinalizeBlock Commit ;`
`ConsensusRounds : ConsensusRound | ConsensusRound ConsensusRounds ;`

So, in this case, `ConsensusHeight` will be `FinalizeBlock Commit`.